### PR TITLE
Restructure start scripts, add empty model mode, add tests for coupled runs, scenario configs, and start functions

### PR DESCRIPTION
**Pull request by mikapfl**
# Purpose of this PR

Lay the groundwork so that future changes to the start scripts can efficiently iterate towards the vision formulated in #1122 :
* move functions from `start.R` and `prepare_and_run.R` into separate functions.
* rename a little for camelCase (just because I was touching everything already)
* make `prepareAndRun.R` sourcable without running the function.
* source the whole `scripts/start/` folder in `start.R` and `prepareAndRun.R` so that all functions are always available
* equally, source everything in `scripts/start` in `start_bundle_coupled.R` and `start_coupled.R` to enable easier code sharing between normal start scripts and coupled start scripts.
* add tests for coupled start scripts
* add tests for scenario config consistency (some tests still disabled because of existing inconsistencies)
* add tests for several individual start functions
* update all documentation regarding these changes

The idea is that if you want to do any refactoring in future, e.g. turning some part of `start.R` into its own function, you can simply add a file named after the new function to `scripts/start/` and your new function will be immediately available everywhere within the start scripts (it will be available from `start.R`, from `prepareAndRun.R`, and from all other functions in `scripts/start/` without needing explicit sourcing).

For me, it was today easier to make this pull request than figuring out how to source a new python-related function I wanted to introduce in REMIND starting scripts such that it is available from `prepare_and_start.R` and `start.R`. After the restructuring this is trivial. Therefore, I think this restructuring will help a lot in future changes, also for the temporary folder work ahead of us

## Type of change

- [x] Refactoring
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the coding etiquette
- [x] I have performed a self-review of my own code
- [x] Changes are commented, particularly in hard-to-understand areas
- [x] I have updated the in-code documentation
- [x] I have adjusted reporting where it was needed
- [x] All automated model tests pass (`FAIL 0` in the output of `make test`)
- [x] Automated coupling tests pass (`FAIL 0 ` in the output of `make test-coupled`)

### CODE REVIEW

**Review by LaviniaBaumstark — COMMENTED** [review]
thanks, nice step towards new structure

**Comment by mikapfl** [comment]
We are working towards testing the coupled system remind-magpie as well, in particular to test the coupled starting scripts.
As a first step, I added an empty model mode, where all the modelling is short-circuited and the input.gdx is just copied to fulldata.gdx. Works nicely, takes less than 10 minutes on my laptop. It is basically a reverse idea compare to `quick`, since in quick, the model is run (in one region only and with loose convergence criteria) but no output scripts are run, and in the empty_model mode the normal reporting is run but not the model.

**Comment by orichters** [comment]
- I moved reading and checking scenario config files to a dedicated function.
- added tests for all config files (currently fails 8 times because unknown switches are used)
- used `#` to out-comment comments in config files, instead of just using `start = 0` and long `____whatever____` names
- deleted [config/scenario_config_ToR_H12.csv](https://github.com/remindmodel/remind/pull/1123/commits/3e786a59afda52343c37212fd6d3b381b0b8a354#diff-1777aa8e14adaef8afab786142d9f3061a8f53b9097c7406745dd43a16cacab8) which created errors, and as it tries to read `config/input.gdx`, it doesn't seem to be usable anyhow.
- implement default for `choose_slurmConfig` based on flags
- adjust `combine_slurmConfig` slightly, so it doesn't fail if I reprepare with `./start.R -R` a `testOneRegi` run that was created by `make test`. Still, it sticks to `--wait` at the moment and the restarted run also waits for it to finish, maybe we want to change that.

**Comment by orichters** [comment]
- deleted `path_gdx_opt` from all `scenario_config_coupled*.csv`, a feature that was deleted by @dklein-pik three years ago in https://github.com/remindmodel/remind/pull/27
- improved tests for `readCheckScenarioConfig()` by adding an error-loaden csv file and see whether it catches them all

**Comment by orichters** [comment]
Six scenario config files still contain
- BioImportTax_EU
- cm_BioImportTax_EU
- cm_bioprod_histlim
- cm_biotrade_phaseout
- cm_build_costDecayStart
- cm_buildings_scen
- cm_CO2priceRegConvYr
- cm_INCONV_PENALTY_bioSwitch
- cm_shSynTrans
that don't exist in REMIND (anymore).

File: config/scenario_config_coupled_NGFS_v3.csv [hunk 1]
```diff
@@ -1,54 +1,54 @@
...
+# d_delfrag;0;;;medium;SSP2|NDC;y2030;2;;;
+# d_rap;0;;;medium;SSP2|NDC;y2020;2;;;
+# SSP2-Base_bIt;0;;;medium;SSP2|NPI;y2150;;;;
+# h_ndc_bIt;0;;;medium;SSP2|NDC;y2150;2;;;
+# h_cpol_bIt;0;;;medium;SSP2|NPI;y2150;2;;;
+# o_1p5c_bIt;0;;;medium;SSP2|NDC;y2020;2;;;
+# o_2c_bIt;0;;;medium;SSP2|NDC;y2020;2;;;
+# d_delfrag_bIt;0;;;medium;SSP2|NDC;y2030;2;;;
+# d_rap_bIt;0;;;medium;SSP2|NDC;y2020;2;;;
+SSP2-Base_bIT;bIT;SSP2-Base_bIT;;priority;SSP2|NPI;y2150;;;;
```
Comment (mikapfl) [inline config/scenario_config_coupled_NGFS_v3.csv:16]: what does `start = bIT` even do?

Comment (orichters) [reply config/scenario_config_coupled_NGFS_v3.csv:16]: If you run `./start.R config/scenario_config_NGFS_v3.csv`, the scripts asks you which runs you want to start, and allows to start the `bIT` group or some other of these groups. This happens if either no scenario has `start = 1`, or if you pass the `--interactive` flag (say: `./start.R --interactive config/scenario_config_rev3p0p0.csv`. But you are right, it should be explained in the tutorial, I guess.

Comment (orichters) [reply config/scenario_config_coupled_NGFS_v3.csv:16]: see https://github.com/remindmodel/remind/pull/705

Comment (mikapfl) [reply config/scenario_config_coupled_NGFS_v3.csv:16]: Thanks for the explanation!

**Comment by orichters** [comment]
- I added this use of `start` in the tutorial
- fixed an error in the counting of errors in `start.R` which was created because I harmonized the name in `start.R` and `start_bundle_coupled.R`. Quite a meta error and maybe a sign that we should avoid this `<<-` construct, in particular now where it isn't in the same file anymore.

**Comment by mikapfl** [comment]
> maybe a sign that we should avoid this <<- construct, in particular now where it isn't in the same file anymore.

Definitely, we should avoid it precisely because it is hard to follow. I even think that it was not at all easier to follow in a single file instead of multiple files, because a single super long file is very hard to follow.

File: start_coupled.R [hunk 2]
```diff
@@ -283,22 +294,17 @@ start_coupled <- function(path_remind, path_magpie, cfg_rem, cfg_mag, runname, m
...
-        " --mail-type=END --comment=REMIND-MAgPIE --tasks-per-node=", nr_of_regions,
-        " --wrap=\"Rscript start_coupled.R coupled_config=", RData_file, "\"")
+        " --mail-type=END --comment=REMIND-MAgPIE --tasks-per-node=", subseq.env$nr_of_regions,
+        " ", subseq.env$sbatch, " --wrap=\"Rscript start_coupled.R coupled_config=", RData_file, "\"")
         message(subsequentcommand)
         if (length(needfulldatagdx) > 0) {
-          system(subsequentcommand)
+          exitCode <- system(subsequentcommand)
+          if (0 < exitCode) {
+            stop("sbatch command failed, check logs")
```
Comment (orichters) [inline start_coupled.R:306]: What is the rational here to fail after the `sbatch` of a single subsequent run failed? Why shouldn't it try to start the other subsequent runs and do the post-processing? I would strongly advise to use an `errorsfound` approach here that only fails really at the end of the script.

File: start_bundle_coupled.R [hunk 3]
```diff
@@ -478,24 +461,35 @@ for(scen in common){
...
     }
     foldername <- file.path("output", fullrunname)
-    if ((i > start_iter_first | !start_magpie) && file.exists(foldername)) {
+    if ((i > start_iter_first || !scenarios_coupled[scen, "start_magpie"]) && file.exists(foldername)) {
       if (! "--test" %in% flags) unlink(foldername, recursive = TRUE, force = TRUE)
       message("Delete ", foldername, if ("--test" %in% flags) " if not in test mode", ". ", appendLF = FALSE)
       deletedFolders <- deletedFolders + 1
     }
+
+    if (cfg_rem$gms$optimization == "nash" && cfg_rem$gms$cm_nash_mode == "parallel") {
```
Comment (orichters) [inline start_bundle_coupled.R:470]: maybe: rename `number_of_regions` to `numberOfNodes` or so? Because it is not the number of regions, but rather the number of CPUs to be used (and, potentially, for your test environment, you might want to set that to 1 as well).

File: start_bundle_coupled.R [hunk 4]
```diff
@@ -117,8 +115,24 @@ if (length(argv) > 0) {
...
+  path_settings_remind  <- sub("scenario_config_coupled", "scenario_config", path_settings_coupled)
+  message("")
 }
 
+message("path_remind:           ", if (dir.exists(path_remind)) green else red, path_remind, NC)
+message("path_magpie:           ", if (dir.exists(path_magpie)) green else red, path_magpie, NC)
+message("path_settings_coupled: ", if (file.exists(path_settings_coupled)) green else red, path_settings_coupled, NC)
+message("path_settings_remind:  ", if (file.exists(path_settings_remind)) green else red, path_settings_remind, NC)
+message("path_remind_oldruns:   ", if (dir.exists(path_remind_oldruns)) green else red, path_remind_oldruns, NC)
+message("path_magpie_oldruns:   ", if (dir.exists(path_remind_oldruns)) green else red, path_magpie_oldruns, NC)
```
Comment (orichters) [inline start_bundle_coupled.R:131]: small error here: must be `path_magpie_oldruns` in the `dir.exists()` part. Is probably an old error…

File: start_coupled.R (hunk 2 above)
Comment (mikapfl) [reply start_coupled.R:306]: I think the rational is:

1. In production runs: if `sbatch` fails, slurm is broken, all bets are off, we might as well stop.
2. In test runs: if `sbatch --wait` fails, the actual run somehow broke, that means the test should fails as fast as possible.

Does that make sense?

Comment (orichters) [reply start_coupled.R:306]: @1: I have no idea how this can happen, but I've heard of temporary cluster hiccups. So I would rather prefer to at least try to start the other runs, because everything that will be started somewhere in the night can save you hours.
@2: That makes sense. Can we use
```
if (0 < exitCode) {
    message("sbatch command failed, check logs")
    stopifnot(! grepl("--wait", subsequentcommand))
}
```
or something like that? And, by the way: which logs are you supposed to check, actually?

Comment (mikapfl) [reply start_coupled.R:306]: all of the logs. Of all runs that were started. And the logs point nicely at each other, the output of the test points at the Base-rem-1 log, this log points at the NDC-rem-1 and Base-rem-2 log etc.

Comment (mikapfl) [reply start_coupled.R:306]: > So I would rather prefer to at least try to start the other runs, because everything that will be started somewhere in the night can save you hours.

I'm not sure, it adds quite some complexity for something which I have never seen and nobody told me they have ever seen it (sbatch fails, but the *next* sbatch will not fail).

**Comment by orichters** [comment]
<details>
  <summary>To see all changes in the scenario config files, click here</summary>

```
File comparison: ../remind-develop/config/scenario_config_AMT.csv -> config/scenario_config_AMT.csv

File comparison: ../remind-develop/config/scenario_config_coupled.csv -> config/scenario_config_coupled.csv
Columns deleted: path_gdx_opt

File comparison: ../remind-develop/config/scenario_config_coupled_GCS.csv -> config/scenario_config_coupled_GCS.csv
Columns deleted: path_gdx_opt

File comparison: ../remind-develop/config/scenario_config_coupled_NGFS_v3.csv -> config/scenario_config_coupled_NGFS_v3.csv
Columns deleted: path_gdx_opt
Changes in the scenarios:
~ SSP2-Base_bIT:
    start: 1 -> bIT
~ h_ndc_bIT:
    start: 1 -> bIT
~ h_cpol_bIT:
    start: 1 -> bIT
~ o_1p5c_bIT:
    start: 1 -> bIT
~ o_2c_bIT:
    start: 1 -> bIT
~ d_delfrag_bIT:
    start: 1 -> bIT
~ d_rap_bIT:
    start: 1 -> bIT
- h_ndc was deleted.
- h_cpol was deleted.
- o_1p5c was deleted.
- o_2c was deleted.
- d_delfrag was deleted.
- d_rap was deleted.
- SSP2-Base_bIt was deleted.
- h_ndc_bIt was deleted.
- h_cpol_bIt was deleted.
- o_1p5c_bIt was deleted.
- o_2c_bIt was deleted.
- d_delfrag_bIt was deleted.
- d_rap_bIt was deleted.

File comparison: ../remind-develop/config/scenario_config_coupled_SHAPE.csv -> config/scenario_config_coupled_SHAPE.csv
Columns deleted: path_gdx_opt

File comparison: ../remind-develop/config/scenario_config_coupled_SSPSDP.csv -> config/scenario_config_coupled_SSPSDP.csv
Columns deleted: path_gdx_opt

File comparison: ../remind-develop/config/scenario_config.csv -> config/scenario_config.csv

File comparison: ../remind-develop/config/scenario_config_DeepEl.csv -> config/scenario_config_DeepEl.csv
Renamed columns: cm_bioenergy_tax -> cm_bioenergy_SustTax
Changes in the scenarios:
- ___Transport_Industry___ was outcommented.

File comparison: ../remind-develop/config/scenario_config_EDGE-T_NDC_NPi_pkbudget.csv -> config/scenario_config_EDGE-T_NDC_NPi_pkbudget.csv

File comparison: ../remind-develop/config/scenario_config_ENGAGE_4p5.csv -> config/scenario_config_ENGAGE_4p5.csv
Columns deleted: cm_esubGrowth
Renamed columns: cm_bioenergy_tax -> cm_bioenergy_SustTax

File comparison: ../remind-develop/config/scenario_config_GCS.csv -> config/scenario_config_GCS.csv

File comparison: ../remind-develop/config/scenario_config_NAVIGATE_300.csv -> config/scenario_config_NAVIGATE_300.csv

File comparison: ../remind-develop/config/scenario_config_NGFS_v3.csv -> config/scenario_config_NGFS_v3.csv
Renamed columns: cm_bioenergy_tax -> cm_bioenergy_SustTax

File comparison: ../remind-develop/config/scenario_config_SHAPE.csv -> config/scenario_config_SHAPE.csv

File comparison: ../remind-develop/config/scenario_config_tradeCap_standalone.csv -> config/scenario_config_tradeCap_standalone.csv

File comparison: ../remind-develop/config/21_regions_EU11/scenario_config_21_EU11_ARIADNE.csv -> config/21_regions_EU11/scenario_config_21_EU11_ARIADNE.csv
Changes in the scenarios:
- _____Calibration_____ was outcommented.
- _____Runs for first years____ was outcommented.
- ____ Standard Scenarios______ was outcommented.
- ____ New Policy Scenarios___ was outcommented.
- _____Technology Scenarios_____ was outcommented.
- ___ Tests___ was outcommented.

File comparison: ../remind-develop/config/21_regions_EU11/scenario_config_21_EU11_ECEMF.csv -> config/21_regions_EU11/scenario_config_21_EU11_ECEMF.csv
Renamed columns: cm_bioenergy_tax -> cm_bioenergy_SustTax
Changes in the scenarios:
- _____REMIND 3.0.0_____ was outcommented.
- _____Calibration_R3_____ was outcommented.
- _____default_R3_____ was outcommented.
- _____pure_carbon_pricing_R3_____ was outcommented.
- _____GHG_target_scenarios_R3_____ was outcommented.
- _____tech_constraint_scenarios_R3_____ was outcommented.
- _____paradigm_shift_scenarios_R3_____ was outcommented.
- ___tests_R3 was outcommented.
- _________________________________ was outcommented.
- _____REMIND 3.0.0 cplx_____ was outcommented.
- _____Calibration_cplx_____ was outcommented.
- _____default_____ was outcommented.
- _____pure_carbon_pricing_____ was outcommented.
- _____GHG_target_scenarios_____ was outcommented.
- _____tech_constraint_scenarios_____ was outcommented.
- _____paradigm_shift_scenarios_____ was outcommented.
- ___oldScenarios was outcommented.
- ___tests was outcommented.
- ________________________________ was outcommented.
- _____WP2 cplx_____ was outcommented.
- ___WP2_Boundary_Condition_scenarios_____ was outcommented.
- ___WP2_Sensitivity_scenarios_____ was outcommented.
- ___tests_WP2 was outcommented.

File comparison: ../remind-develop/config/21_regions_EU11/scenario_config_21_EU11_Fit_for_55_sensitivity.csv -> config/21_regions_EU11/scenario_config_21_EU11_Fit_for_55_sensitivity.csv
Changes in the scenarios:
- _____Calibration_cplx_____ was outcommented.
- _____default_____ was outcommented.
- ______ref_Eff_Target______ was outcommented.
- ______exploration_scenarios______ was outcommented.
- ______SENSITIVITY_SCENARIOS_______ was outcommented.
- _____biomass_____ was outcommented.
- _____Capacity replacement_____ was outcommented.
- _____Energy efficiency 2030_____ was outcommented.
- _____Global adjustment cost_____ was outcommented.
- ______nonICE_______ was outcommented.
- _____Electric vehicles push_____ was outcommented.
- _____bev_plus_LDV_sales_____ was outcommented.
- _____Hydrogen push_____ was outcommented.
- _____Synfuels push_____ was outcommented.
- _____CCS_____ was outcommented.
- _____Nuclear_____ was outcommented.
- _____Renewables (wind and solar)_____ was outcommented.
- _____DAC_____ was outcommented.
- _____World Mitigation Effort_____ was outcommented.
- _____Storage and flexibility requirements_____ was outcommented.
- ___test was outcommented.

File comparison: ../remind-develop/config/21_regions_EU11/scenario_config_21_EU11_SSPSDP.csv -> config/21_regions_EU11/scenario_config_21_EU11_SSPSDP.csv

File comparison: ../remind-develop/config/tests/scenario_config_compile.csv -> config/tests/scenario_config_compile.csv

../remind-develop/config/tests/scenario_config_coupled_shortCascade.csv not found.

File comparison: ../remind-develop/config/tests/scenario_config_default.csv -> config/tests/scenario_config_default.csv

../remind-develop/config/tests/scenario_config_empty.csv not found.

File comparison: ../remind-develop/config/tests/scenario_config_quick.csv -> config/tests/scenario_config_quick.csv

../remind-develop/config/tests/scenario_config_shortCascade.csv not found.
```
</details>

File: start_coupled.R (hunk 2 above)
Comment (mikapfl) [reply start_coupled.R:306]: Like, if we think this happens, I'd rather but in some logic which re-tries three times every `sbatch` call, it would actually solve things in this situation (sbatch fails, but the next sbatch will be fine)

Comment (orichters) [reply start_coupled.R:306]: Ok, but even if `sbatch` fails, I would actually prefer to complete the runtime and compareScenario stuff that is below. Additionally, in my short history with REMIND and the cluster, I have seen many things nobody has seen before (thinking about `.lock`).

File: scripts/start/readCheckScenarioConfig.R [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  if (any(containsdot)) {
+    warning("These titles contain a dot: ",
+            paste0(rownames(scenConf)[containsdot], collapse = ", "),
+            " – GAMS would not tolerate this, and quit working at a point where you least expect it. Stopping now.")
+  }
+  underscore <- grep("_$", rownames(scenConf))
+  if (any(underscore)) {
+    warning("These titles end with _: ",
+            paste0(rownames(scenConf)[underscore], collapse = ", "),
+            ". This may lead to wrong gdx files being selected. Stopping now.")
```
Comment (orichters) [inline scripts/start/readCheckScenarioConfig.R:40]: Many csv used `____whatever____` and `start=0` as "comment". Should we maybe suggest in line 40 that they please properly outcomment using `#`?

File: start_coupled.R (hunk 2 above)
Comment (orichters) [reply start_coupled.R:306]: And: If you use `qos=shorty` or something like that, `sbatch --qos=shorty --wrap="Rscript start.R --test"` fails without an actual problem with `sbatch`. So you can trigger that with a simple typo in your config.

Comment (mikapfl) [reply start_coupled.R:306]: I guess if you have scenarios with different qos, then the script does something useful after failing to start another scenario with a different qos, yes. I'm not sure how useful given that it is missing some scenarios, but I guess you really rather debug slightly weird behaviour because one scenario is missing in the chain and re-start after fixing if that means the re-start is faster.

File: scripts/start/runGamsCompile.R [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#' compile a modelFile with a cfg 
+#'
+#' @param modelFile filename of model file to be compiled
+#' @param cfg list with REMIND configuration
+#' @param interactive boolean, if TRUE, will ask user to compile again after fails
+#' @author Oliver Richters
+#' @return boolean whether compilation was successful
+runGamsCompile <- function(modelFile, cfg, interactive = TRUE) {
```
Comment (orichters) [inline scripts/start/runGamsCompile.R:8]: I remember that @0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q suggested that `--gamscompile` should run `prepare()` to download input data. Can this easily be done now, or would this be a bigger change?

File: start_coupled.R [hunk 7]
```diff
@@ -283,22 +296,18 @@ start_coupled <- function(path_remind, path_magpie, cfg_rem, cfg_mag, runname, m
...
-        " --wrap=\"Rscript start_coupled.R coupled_config=", RData_file, "\"")
+        " --mail-type=END --comment=REMIND-MAgPIE --tasks-per-node=", subseq.env$numberOfTasks,
+        " ", subseq.env$sbatch, " --wrap=\"Rscript start_coupled.R coupled_config=", RData_file, "\"")
         message(subsequentcommand)
         if (length(needfulldatagdx) > 0) {
-          system(subsequentcommand)
+          exitCode <- system(subsequentcommand)
+          if (0 < exitCode) {
+            message("sbatch command failed, check logs")
+            stopifnot(! grepl("--wait", subsequentcommand))
```
[the remark this answers is not in the document]
Comment (orichters) [reply start_coupled.R:309]: I changed the behavior slightly here: In a "normal" runs (so no test), it will fail, but only after running until the end of the function, to make sure that the original run is really finished and it tried to start all subsequent runs.
https://github.com/mikapfl/remind/pull/13/files#diff-8ff4522c53f53332c37529587e7f7e9b7227c40c08b9658e9133c11a442b4612

**Review by orichters — APPROVED** [review]
I reviewed the code again and fixed some small typos etc. in [this PR to Mikas code](https://github.com/mikapfl/remind/pull/13). I'm fine with the current state that certainly can be improved further in the future, thanks a lot for this effort!

File: scripts/start/runGamsCompile.R (hunk 6 above)
Comment (mikapfl) [reply scripts/start/runGamsCompile.R:8]: I think it would be rather easy, but let's do it in the next restructuring PR :wink:

File: start_coupled.R (hunk 7 above)
[the remark this answers is not in the document]
Comment (mikapfl) [reply start_coupled.R:314]: I added an explanatory message.

File: scripts/start/readCheckScenarioConfig.R [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+            ". This may lead to wrong gdx files being selected. Please use # to make comments, not underscores.",
+            " Stopping now.")
+  }
+  if ("path_gdx_ref" %in% names(scenConf) && ! "path_gdx_refpolicycost" %in% names(scenConf)) {
+    scenConf$path_gdx_refpolicycost <- scenConf$path_gdx_ref
+    message("In ", filename,
+        ", no column path_gdx_refpolicycost for policy cost comparison found, using path_gdx_ref instead.")
+  }
+  errorsfound <- sum(toolong) + sum(regionname) + sum(containsdot) + sum(underscore)
+  scenConf[, names(path_gdx_list)[! names(path_gdx_list) %in% names(scenConf)]] <- NA
```
Comment (dklein-pik) [inline scripts/start/readCheckScenarioConfig.R:49]: `path_gdx_list` is accessed without being supplied explicitly. However, in the new structure it is sourced together with the scripts if I am not mistaken. So we use it like a global constant now?

**Review by dklein-pik — APPROVED** [review]

File: scripts/start/readCheckScenarioConfig.R (hunk 8 above)
Comment (orichters) [reply scripts/start/readCheckScenarioConfig.R:49]: In a way, yes. But is true that you cannot source only this file here and run `readCheckScenarioConfig`, without also sourcing `path_gdx_list`. Not sure whether that is a problem.

Comment (dklein-pik) [reply scripts/start/readCheckScenarioConfig.R:49]: `readCheckScenarioConfig` seems to be a quite specific function . Thus, I would suggest to keep it as is and change it if we observe that `readCheckScenarioConfig` is used elsewhere.

Comment (orichters) [reply scripts/start/readCheckScenarioConfig.R:49]: A `if (! exists("path_gdx_list")) stop("You have to source path_gdx_list.R as well!")` wouldn't cost us much, on the other hand. Same for `configureCfg`.
